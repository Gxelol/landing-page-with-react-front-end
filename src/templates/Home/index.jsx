import { useEffect, useRef, useState } from 'react';
import { useLocation } from 'react-router-dom';

import { mapData } from '../../api/map-data';

import { GridTwoColumns } from '../../components/GridTwoColumns';
import { GridContent } from '../../components/GridContent';
import { GridText } from '../../components/GridText';
import { GridImage } from '../../components/GridImage';

import { Base } from '../Base';
import { PageNotFound } from '../PageNotFound';
import { Loading } from '../Loading';

function Home() {
  const [data, setData] = useState([]);
  const location = useLocation();

  useEffect(() => {
    const pathname = location.pathname.replace(/[^a-z0-9-_]/gi, '');
    const slug = pathname ? pathname : 'landing-page';

    const load = async () => {
      try {
        const data = await fetch(
          `http://localhost:1337/api/pages?populate=deep&pagination[pageSize]=1&sort[0]=id:desc`,
        );
        const json = await data.json();
        const { attributes } = json.data[0];
        const pageData = mapData([attributes]);
        setData(() => pageData[0]);
      } catch {
        setData(undefined);
      }
    };

    load();
  }, [location]);

  if (data === undefined) {
    return <PageNotFound />;
  }

  if (data && !data.slug) {
    return <Loading />;
  }

  const { menu, sections, footerHtml } = data;
  const { links, text, link, srcImg } = menu;

  return (
    <Base
      links={links}
      footerHtml={footerHtml}
      logoData={{ text, link, srcImg }}
    >
      {sections.map((section, index) => {
        const { component } = section;
        const key = `${component}-${index}`;

        if (component === 'section.section-two-columns') {
          return <GridTwoColumns key={key} {...section} />;
        }

        if (component === 'section.section-content') {
          return <GridContent key={key} {...section} />;
        }

        if (component === 'section.section-grid-text') {
          return <GridText key={key} {...section} />;
        }

        if (component === 'section.section-grid-image') {
          return <GridImage key={key} {...section} />;
        }
      })}
    </Base>
  );
}

export default Home;
