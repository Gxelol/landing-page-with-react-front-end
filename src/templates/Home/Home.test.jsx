import { renderTheme } from '../../styles/renderTheme.js';
import Home from '.';

describe('<Home />', () => {
  it('should render <Home />', () => {
    renderTheme(<Home />);
  });
});
