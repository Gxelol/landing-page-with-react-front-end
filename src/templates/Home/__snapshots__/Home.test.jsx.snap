// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
.c1 {
  color: #0A1128;
}

.c0 {
  background: #dc143c;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Hello World!
  </h1>
</div>
`;
